src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -28,10 +29,19 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
             // destroy the data structures representing outstanding progress records
             // take down and destroy the progress display
 
+            if (_progPaneUpdateTimer != null)
+            {
+                // Stop update a progress pane and destroy timer
+                _progPaneUpdateTimer.Dispose();
+                _progPaneUpdateTimer = null;
+            }
+            progPaneUpdateFlag = false;
+
             if (_progPane != null)
             {
                 Dbg.Assert(_pendingProgress != null, "How can you have a progress pane and no backing data structure?");
 
+                // lock (_instanceLock) is not needed here because lock is done at global level
```

Comment:
I lost the lock for the three months 😄
There is three points to call `ResetProgress`:
1. https://github.com/PowerShell/PowerShell/blob/9e27f4a64c03f2059a0fd9ba04c3d5c1363af92f/src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs#L2583
2. https://github.com/PowerShell/PowerShell/blob/02b5f357a20e6dee9f8e60e3adb9025be3c94490/src/Microsoft.PowerShell.ConsoleHost/host/msh/Executor.cs#L266
3. https://github.com/PowerShell/PowerShell/blob/02b5f357a20e6dee9f8e60e3adb9025be3c94490/src/Microsoft.PowerShell.ConsoleHost/host/msh/Executor.cs#L420

All without a lock. It seems we need add ` lock (_instanceLock) ` in  ResetProgress.